airbyte-integrations/connectors/destination-duckdb/destination_duckdb/destination.py

```diff
@@ -149,14 +193,17 @@ def _safe_write(*, con: duckdb.DuckDBPyConnection, buffer: Dict[str, Dict[str, L
             logger.exception(
                 f"Writing with pyarrow view failed, falling back to writing with executemany. Expect some performance degradation."
             )
+            column_names_list = buffer[stream_name].keys()
+            column_names = ", ".join(column_names_list.keys())
+            params = ", ".join(["?"] * len(column_names_list))
             query = f"""
             INSERT INTO {schema_name}.{table_name}
-                (_airbyte_ab_id, _airbyte_emitted_at, _airbyte_data)
-            VALUES (?,?,?)
+                ({column_names})
+            VALUES ({params})
```

Comment:
Note: If we are in dedupe mode, we'll need this to be a logical merge upsert rather than an insert.

Reply:
(note to self)
```python
class DestinationSyncMode(Enum):
    append = 'append'
    overwrite = 'overwrite'
    append_dedup = 'append_dedup'
```

Reply:
OK, this has been addressed! I have migrated over the SQL processor from PyAirbyte and I'm reusing the class to create a temp table, deduplicate it by primary key and insert to the final table using `INSERT AND REPLACE` if the `append_dedup` sync mode is selected.